janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+            final StandardJanusGraph graph2 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void shouldCreateTwoGraphsUsingSameTemplateConfiguration() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            final StandardJanusGraph graph2 = (StandardJanusGraph) ConfiguredGraphFactory.create("graph2");
+
+            assertNotNull(graph1);
+            assertNotNull(graph2);
+
+            assertEquals(graph1.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
+            assertEquals(graph2.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph2");
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            configGraphManagement.removeConfiguration("graph2");
+            ConfiguredGraphFactory.close("graph1");
+            ConfiguredGraphFactory.close("graph2");
+        }
+    }
+
+    @Test
+    public void ensureCallingGraphCloseResultsInNewGraphReferenceOnNextCallToOpen() throws Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            assertNotNull(graph);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
+            graph.close();
+            assertTrue(graph.isClosed());
+
+            final StandardJanusGraph newGraph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+
+            assertFalse(newGraph.isClosed());
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void fileToConfigurationShouldReturnAllProperties() throws IOException, ConfigurationException {
+        final File file = new File(getClass().getClassLoader().getResource("org/janusgraph/core/janusgraph-testing.properties").getFile());
+        final String propertiesFileLocation = file.getAbsolutePath();
+        final Configuration config = ConfiguredGraphFactory.fileToConfiguration(propertiesFileLocation);
+
+        assertNotNull(config);
+        assertEquals(Iterators.size(config.getKeys()), 2);
```

Comment:
Expected values, i.e., constants, are the first parameter to `assertEquals()`, the computed value is second. Similar comment for the following 2 lines as well — please reverse the argument order and put constants first.

They are the same (since it's equality) and it will pass or fail just the same, but the error message will not make any sense, because it will say:

> Expected: (the computed value), actual: 2

and the developer will spend some time trying to debug why their computation is wrong, when in fact, it's just that the parameters to `assertEquals` are reversed.

More info:

* https://stackoverflow.com/questions/16267660/assert-assertequals-junit-parameters-order